web/clusters.go

```diff
@@ -11,6 +12,239 @@ import (
 	"github.com/trento-project/trento/internal/hosts"
 )
 
+type Node struct {
+	Name       string
+	Attributes map[string]string
+	Resources  []*Resource
+	Ip         string
+	Health     string
+	VirtualIps []string
+}
+
+type Resource struct {
+	Id        string
+	Type      string
+	Role      string
+	Status    string
+	FailCount int
+}
+
+type Nodes []*Node
+
+const (
+	ClusterTypeScaleUp  = "HANA scale-up"
+	ClusterTypeScaleOut = "HANA scale-out"
+	ClusterTypeUnknown  = "Unknown"
+)
+
+func detectClusterType(c *cluster.Cluster) string {
+	var hasSapHanaTopology, hasSAPHanaController, hasSAPHana bool
+
+	for _, c := range c.Crmmon.Clones {
+		for _, r := range c.Resources {
+			switch r.Agent {
+			case "ocf::suse:SAPHanaTopology":
+				hasSapHanaTopology = true
+			case "ocf::suse:SAPHana":
+				hasSAPHana = true
+			case "ocf::suse:SAPHanaController":
+				hasSAPHanaController = true
+			}
+		}
+	}
+
+	switch {
+	case hasSapHanaTopology && hasSAPHana:
+		return ClusterTypeScaleUp
+	case hasSapHanaTopology && hasSAPHanaController:
+		return ClusterTypeScaleOut
+	default:
+		return ClusterTypeUnknown
+	}
+}
+
+func stoppedResources(c *cluster.Cluster) []*Resource {
+	var stoppedResources []*Resource
+
+	for _, r := range c.Crmmon.Resources {
+		if r.NodesRunningOn == 0 && !r.Active {
+			resource := &Resource{
+				Id: r.Id,
+			}
+			stoppedResources = append(stoppedResources, resource)
+		}
+	}
+
+	return stoppedResources
+}
+
+func (node *Node) Role() string {
+	if r, ok := node.Attributes["hana_prd_roles"]; ok {
```

Comment:
Maybe as a last request. Could we put here, as docstring, the meaining of this field and contents? We will forget otherwise hehe

Reply:
yes! also this should be HANARole I think